rust/rsc/src/rsc/main.rs

```diff
@@ -213,7 +217,55 @@ fn launch_eviction(conn: Arc<DatabaseConnection>, tick_interval: u64, deadline:
                 .await
                 .unwrap();
 
-            tracing::info!(%res.rows_affected, "Performed TTL eviction tick.");
+            tracing::info!(%res.rows_affected, "Performed Job TTL eviction tick.");
+        }
+    });
+}
+
+fn launch_blob_eviction(
+    conn: Arc<DatabaseConnection>,
+    tick_interval: i64,
+    deadline: i64,
+    blob_stores: HashMap<Uuid, Arc<dyn blob::DebugBlobStore + Sync + Send>>,
+) {
+    // TODO: This should probably be a transaction so that a job can't add a new reference to a
+    // blob as we are deleting it.
+
+    tokio::spawn(async move {
+        let mut interval = tokio::time::interval(std::time::Duration::from_secs(
+            tick_interval.try_into().unwrap(),
+        ));
+        loop {
+            interval.tick().await;
+
+            // Blobs must be at least this old to be considered for eviction.
+            // This gives clients time to reference a blob before it gets evicted.
+            let deadline = (Utc::now() - Duration::seconds(deadline)).naive_utc();
+
+            let blobs = blob_store_service::fetch_unreferenced_blobs(conn.as_ref(), deadline)
+                .await
+                .unwrap();
+            let blob_ids: Vec<Uuid> = blobs.iter().map(|blob| blob.id).collect();
+            let mut affected: u64 = blob_ids.len().try_into().unwrap();
+
+            if affected == 0 {
+                tracing::info!(%affected, "Performed Blob TTL eviction tick.");
+                continue;
+            }
+
+            // Delete blobs from database
+            let res = blob_store_service::delete_blobs_by_ids(conn.as_ref(), blob_ids)
+                .await
+                .unwrap();
+            affected = res.rows_affected;
```

Comment:
We want this to be the same variable rather than a `deleted` non mutable variable because of the way the `tracing::info!` macro works?

Reply:
It seems like if those two numbers are deleted we would want to record that actually. It would mean that we had some mismatch in the queries or a partial failure in the deletion.

Reply:
I think it'll be clearer if I just have two `tracing::info!`s here. Once for "found this many to delete" once for "actually deleted this many"